examples/api_test/test_ssspapi.cu

```diff
@@ -0,0 +1,128 @@
+ 
+  //  int num_query_nodes = 3, num_query_edges = 6;
+   
+ 
+  
+   //gunrock::util::Parameters parameters("sssp");
+
+   unsigned int memspace;
+ 
+   if(*argv[1] == '1'){
+     //allocating arrays on CPU. If memspace ==2, need to allocate on gpu
+     memspace = 0x01;
+     
+    //  unsigned long *sm_counts = (unsigned long *)malloc(sizeof(unsigned long));
+    //  unsigned long *list_sm = (unsigned long *)malloc(sizeof(unsigned long)); 
+     float *distances = (float *)malloc(num_nodes * sizeof(float));
+     int *preds = (int*)NULL;//(int *)malloc(sizeof(int));
+
+     double elapsed =
+       sssp(num_nodes, num_edges, row_offsets, col_indices,
+            edge_values, source, mark_pred, distances, preds, memspace);
+     
+     printf("Distances: ");
+     for(int i = 0; i < num_nodes; i++){
+       printf("%f ", distances[i]);
+     }
+     printf("\n");
+     //printf("List of matched subgraphs: [%i]\n", preds[0]);
+   }
+ 
+    if(*argv[1] == '2'){
+ 
+     //TODO: Make sure this parameter change is acceptable.
+     //If yes, make sure this gets represented in sssp_app.cu if memspace != host
+     //parameters.Set<bool>("quiet", true);
+
+     memspace = 0x02;
+ 
+     thrust::device_vector<int> device_row_offsets(row_offsets, row_offsets+8);
+     thrust::device_vector<int> device_col_indices(col_indices, col_indices+15);
+     thrust::device_vector<float> device_edge_values(edge_values, edge_values+15);
+     
+     thrust::device_vector<float> device_distances(num_nodes);
+     thrust::device_vector<int> device_preds(1);
+
+     double elapsed =
+        sssp(num_nodes, num_edges, device_row_offsets.data().get(), device_col_indices.data().get(),
+          device_edge_values.data().get(), source, mark_pred, device_distances.data().get(), device_preds.data().get(), memspace);
+     
+     printf("Distances: ");
+     thrust::copy(device_distances.begin(), device_distances.end(), std::ostream_iterator<int>(std::cout, " "));     
+     //thrust::copy(device_list_sm.begin(), device_list_sm.end(), std::ostream_iterator<int>(std::cout, " "));
+ 
+    }
+
+    // if(*argv[1] == '0'){
+    //   //testing mode
+
+    //   bool quiet = parameters.Get<bool>("quiet");    
+    //   printf("Default quiet parameter value: %d\n", quiet);
+    //   parameters.Set<bool>("quiet", true);
+    //   quiet = parameters.Get<bool>("quiet");    
+    //   printf("Modified quiet parameter value: %d\n", quiet);
+
+    
+    // }
+
+ 
+ 
+ 
+ 
+ 
+ 
+ 
+ //  double elapsed =
+ //       sm(num_data_nodes, num_data_edges, device_data_row_offsets.data().get(), data_col_indices,
+ //          num_query_nodes, num_query_edges, query_row_offsets, query_col_indices,
+ //          1, sm_counts, list_sm, location);
+ 
+ 
+ 
+ 
+   // cudaFree(device_data_row_offsets);
+   //if (sm_counts) free(sm_counts);
+   //if (list_sm) free(list_sm);
+ 
+   return 0;
+ }
+ 
```

Comment:
Lots of comments and dead space. You can even try using the `clang-format` tool to format this file and other files properly.